src/main/java/seedu/address/logic/commands/FindCommand.java

```diff
@@ -48,13 +56,14 @@ public boolean equals(Object other) {
         }
 
         FindCommand otherFindCommand = (FindCommand) other;
-        return predicate.equals(otherFindCommand.predicate);
+        return nPredicate.equals(otherFindCommand.nPredicate) && cPredicate.equals(otherFindCommand.cPredicate);
     }
 
     @Override
     public String toString() {
         return new ToStringBuilder(this)
-                .add("predicate", predicate)
+                .add("name predicate", nPredicate)
+                .add("course predicate", cPredicate)
```

Comment:
The point is that the fieldName should be **the name of the field**, according to `ToStringBuilder`'s documentation.

Also that the field name appears to be needlessly obscure as well, but not a big deal.